_resources/port1.0/group/python-1.0.tcl

```diff
@@ -447,3 +488,33 @@ pre-livecheck {
         livecheck.type  none
     }
 }
+
+# set-up a test phase
+pre-test {
+    if {[option python.test]} {
+        # return for py-<portname> stub ports, which should not have a test-phase enabled
+        if {[string match py-* [option name]] && ${name} eq ${subport}} {
+            return
+        }
+
+        test.run            yes
```

Comment:
Does this really work? (Does `sudo port test ...` run the tests?) I thought `test.run` would have to be `yes` before MacPorts would consider running a test phase. If I'm right, then this directive would have to be outside of the `pre-test` block. Since you don't know what value to set it to until after the port has had a change to set `python.test`, you'd probably need another procedure invocation to set it (something like `default test.run {[python_get_defaults test_run]}`).

Reply:
Even if it does work, I think even having two separate options that mean essentially the same thing is overcomplicating matters. The portfile can set `test.run yes` if tests should be run, and the portgroup can look at `$subport` when deciding how to set `test.cmd` and the dependencies.

Reply:
it certainly works for the Python ports that are part of this PR. But I see the point to just use `test.run` instead of introducing a new variable - seems simpler indeed.